src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/FireUtilitySpell.java

```diff
@@ -0,0 +1,186 @@
+import net.minecraft.client.player.LocalPlayer;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.event.entity.living.LivingAttackEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.DecimalFormat;
+import java.util.List;
+import java.util.Optional;
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class FireUtilitySpell extends PassiveSpellItem {
+    private final int manaCost;
+    private final int aoeRange;
+    private final double damage;
+    private final int damageRate;
+
+
+    public FireUtilitySpell(Properties properties, FireUtilitySpellConfig config) {
+        super(properties, config.LEVEL, MagicType.FIRE, SpellType.UTILITY);
+
+        this.manaCost = config.MANA_COST.get();
+        this.aoeRange = config.AOE_RANGE.get();
+        this.damage = config.DAMAGE.get();
+        this.damageRate = config.DAMAGE_RATE.get();
+    }
+
+
+    @SubscribeEvent
+    public static void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        if (event.phase == TickEvent.Phase.START) {
+            Player player = event.player;
+            List<FireUtilitySpell> equippedSpells = getEquippedSpellItemsOfType(FireUtilitySpell.class, player);
+            if (equippedSpells.size() > 0) {
+                player.getCapability(PlayerUtilityToggleProvider.UTILITY_TOGGLE).ifPresent(utilityToggle -> {
+                    if (utilityToggle.fire) {
+                        FireUtilitySpell highestSpell = getHighestSpellItem(equippedSpells);
+                        int damageRate = highestSpell.damageRate;
+                        Level level = player.getLevel();
+                        if (level.getGameTime() % damageRate == 0) {
+                            int manaCost = 0;
+                            int range = 0;
+                            int damage = 0;
+                            for (FireUtilitySpell spell : equippedSpells) {
+                                manaCost += spell.manaCost;
+                                range += spell.aoeRange;
+                                damage += spell.damage;
+                            }
+
+                            int particleCount = range / 5;
```

Comment:
We might wanna limit particles to line of sight too

Reply:
Done.